keyserver/internal/device_list_update.go

```diff
@@ -186,6 +196,36 @@ func (u *DeviceListUpdater) Start() error {
 	return nil
 }
 
+// cleanUp removes stale device entries for users we don't share a room with anymore
+func (u *DeviceListUpdater) cleanUp() error {
+	if u.rsAPI == nil {
+		return nil
+	}
+	staleUsers, err := u.db.StaleDeviceLists(u.process.Context(), []gomatrixserverlib.ServerName{})
+	if err != nil {
+		return err
+	}
+
+	maxRetries := 3
+
+	// In polylith mode, the roomserver api might not be up yet, so we try again
```

Comment:
It is, indeed. Something I was thinking about - we could implement some retry mechanism in our internal API calls, but this would increase the complexity around them.